### PR TITLE
Changes and Fixes from R-shape commit

### PR DESCRIPTION
**Pull request by nzardosh**

### CODE REVIEW

File: machine_learning_hep/processerdhadrons_jet.py [hunk 1]
```diff
@@ -401,65 +451,69 @@ def process_efficiency_single(self, index):
...
-                    df_mc_reco = pickle.load(openfile(self.mptfiles_recoskmldec[bin_id][index], "rb"))
+                    df_mc_reco = pickle.load(
+                        openfile(self.mptfiles_recoskmldec[bin_id][index], "rb"))
                 df_mc_reco = adjust_nsd(df_mc_reco)
                 if self.s_evtsel is not None:
                     df_mc_reco = df_mc_reco.query(self.s_evtsel)
                 if self.s_jetsel_reco is not None:
-                    if self.p_usejetptbinned_deff is True:
-                        df_mc_reco = df_mc_reco.query(self.s_jetsel_reco)
+                    df_mc_reco = df_mc_reco.query(self.s_jetsel_reco)
```
Comment (vkucera) [inline machine_learning_hep/processerdhadrons_jet.py:488]: Why is the `p_usejetptbinned_deff` condition removed?

File: machine_learning_hep/processerdhadrons_jet.py [hunk 2]
```diff
@@ -401,65 +451,69 @@ def process_efficiency_single(self, index):
...
-                    df_mc_reco = selectdfrunlist(df_mc_reco, \
-                             self.run_param[self.runlistrigger], "run_number")
-                df_mc_gen = pickle.load(openfile(self.mptfiles_gensk[bin_id][index], "rb"))
+                    df_mc_reco = selectdfrunlist(df_mc_reco,
+                                                 self.run_param[self.runlistrigger], "run_number")
+                df_mc_gen = pickle.load(
+                    openfile(self.mptfiles_gensk[bin_id][index], "rb"))
                 df_mc_gen = adjust_nsd(df_mc_gen)
-                if self.p_usejetptbinned_deff is True:
+                if self.s_jetsel_gen is not None:
```
Comment (vkucera) [inline machine_learning_hep/processerdhadrons_jet.py:497]: Why is the `p_usejetptbinned_deff` condition removed?

File: machine_learning_hep/processerdhadrons_jet.py [hunk 3]
```diff
@@ -401,65 +451,69 @@ def process_efficiency_single(self, index):
...
-                        self.var2_low_inc, self.var2_up_inc)
+                    df_mc_reco = seldf_singlevar(df_mc_reco, self.v_var2_binning,
+                                                 self.lvar2_binmin_reco[ibin2], self.lvar2_binmax_reco[ibin2])
+                    df_mc_gen = seldf_singlevar(df_mc_gen, self.v_var2_binning,
+                                                self.lvar2_binmin_reco[ibin2], self.lvar2_binmax_reco[ibin2])  # FIXME use lvar2_binmin_gen pylint: disable=fixme
+                else:
+                    df_mc_reco = seldf_singlevar(df_mc_reco, self.v_var2_binning,
+                                                 self.lvar2_binmin_reco[0], self.lvar2_binmax_reco[-1])
+                    df_mc_gen = seldf_singlevar(df_mc_gen, self.v_var2_binning,
+                                                self.lvar2_binmin_reco[0], self.lvar2_binmax_reco[-1])  # FIXME use lvar2_binmin_gen pylint: disable=fixme
```
Comment (vkucera) [inline machine_learning_hep/processerdhadrons_jet.py:516]: Why this change?

File: machine_learning_hep/processerdhadrons_jet.py [hunk 4]
```diff
@@ -645,21 +718,22 @@ def effcorr_response(self, df_noeffcorr):
...
                 # limit the prompt(nonprompt) df with current bin ranges and
                 # assign corresponding efficiency to each bin as a new df column
-                df_tmp = seldf_singlevar(df_noeffcorr, "pt_cand", \
-                        self.lpt_finbinmin[ipt], self.lpt_finbinmax[ipt])
+                df_tmp = seldf_singlevar(df_noeffcorr, "pt_cand",
+                                         self.lpt_finbinmin[ipt], self.lpt_finbinmax[ipt])
                 df_tmp['eff'] = eff
-                if eff > 0:
-                    df_tmp['weight'] = 1/eff
+                df_tmp['unity'] = 1.0
```
Comment (vkucera) [inline machine_learning_hep/processerdhadrons_jet.py:724]: Is this needed?

File: machine_learning_hep/data/data_prod_20210223/database_variations_LcpK0spp_jet_FF.yml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 categories:
     default:
-        activate: no
+        activate: yes
```
Comment (vkucera) [inline machine_learning_hep/data/data_prod_20210223/database_variations_LcpK0spp_jet_FF.yml:3]:
```suggestion
        activate: no
```

File: machine_learning_hep/data/data_prod_20210223/database_variations_LcpK0spp_jet_FF.yml [hunk 6]
```diff
@@ -265,7 +265,7 @@ categories:
         rms_both_sides: true
         variations:
             cutvar:
-                activate: [yes, no, no, no, no, no, no, no, no, no, no, no, no, no, no, no, no, no, no, no, no, no, no]
+                activate: [yes, no, yes, yes, yes, no, yes, yes, yes, yes, yes, yes, yes, yes, yes, yes, yes, yes, yes, yes, yes, yes, yes]
```
Comment (vkucera) [inline machine_learning_hep/data/data_prod_20210223/database_variations_LcpK0spp_jet_FF.yml:268]: Why are there two `no`s?

File: machine_learning_hep/processerdhadrons_jet.py [hunk 7]
```diff
@@ -349,50 +391,58 @@ def process_histomass_single(self, index):
...
                     h_invmass_refl.Write()
 
-
     # pylint: disable=line-too-long
     def process_efficiency_single(self, index):
         out_file = TFile.Open(self.l_histoeff[index], "recreate")
 
-        cut_d0y = "abs(y_cand) < 0.8 and abs(z_vtx_gen) < 10" # FIXME Fix this via the database. pylint: disable=fixme
+        # FIXME Fix this via the database. pylint: disable=fixme
+        cut_d0y = "abs(y_cand) < 0.8 and abs(z_vtx_gen) < 10"
```
Comment (vkucera) [inline machine_learning_hep/processerdhadrons_jet.py:399]: This has to be adjusted too.

File: machine_learning_hep/data/data_prod_20210223/database_variations_LcpK0spp_jet_FF.yml (hunk 5 above)
Comment (nzardosh) [reply machine_learning_hep/data/data_prod_20210223/database_variations_LcpK0spp_jet_FF.yml:3]: ok

File: machine_learning_hep/data/data_prod_20210223/database_variations_LcpK0spp_jet_FF.yml (hunk 6 above)
Comment (nzardosh) [reply machine_learning_hep/data/data_prod_20210223/database_variations_LcpK0spp_jet_FF.yml:268]: the fits seemed unstable with these so i removed them

File: machine_learning_hep/processerdhadrons_jet.py (hunk 1 above)
Comment (nzardosh) [reply machine_learning_hep/processerdhadrons_jet.py:434]: i think we should do this regardless of whether we do the per pT efficiency or not cause i dont think its a pT dependent cut

File: machine_learning_hep/processerdhadrons_jet.py (hunk 2 above)
Comment (nzardosh) [reply machine_learning_hep/processerdhadrons_jet.py:442]: same as above

File: machine_learning_hep/processerdhadrons_jet.py (hunk 4 above)
Comment (nzardosh) [reply machine_learning_hep/processerdhadrons_jet.py:669]: i think it is removed now

File: machine_learning_hep/processerdhadrons_jet.py [hunk 8]
```diff
@@ -424,42 +424,41 @@ def process_efficiency_single(self, index):
...
                     df_mc_reco = selectdfrunlist(df_mc_reco, \
-                             self.run_param[self.runlistrigger], "run_number")
+                        self.run_param[self.runlistrigger], "run_number")
                 df_mc_gen = pickle.load(openfile(self.mptfiles_gensk[bin_id][index], "rb"))
                 df_mc_gen = adjust_nsd(df_mc_gen)
-                if self.p_usejetptbinned_deff is True:
+                if self.s_jetsel_gen is not None:
                     df_mc_gen = df_mc_gen.query(self.s_jetsel_gen)
-                df_mc_gen = df_mc_gen.query(cut_d0y)
+                df_mc_gen = df_mc_gen.query(self.s_jetsel_gen)
```
Comment (vkucera) [inline machine_learning_hep/processerdhadrons_jet.py:444]: `s_jetsel_gen` is applied twice here, once under condition, second time regardless.

Comment (vkucera) [reply machine_learning_hep/processerdhadrons_jet.py:444]: Fixed in https://github.com/nzardosh/MachineLearningHEP/pull/1

File: machine_learning_hep/processerdhadrons_jet.py (hunk 4 above)
Comment (vkucera) [reply machine_learning_hep/processerdhadrons_jet.py:669]: Fixed in https://github.com/nzardosh/MachineLearningHEP/pull/1

File: machine_learning_hep/processerdhadrons_jet.py (hunk 1 above)
Comment (vkucera) [reply machine_learning_hep/processerdhadrons_jet.py:434]: It affects the effective `y_cand`-averaged efficiency through the `eta_jet` cut.

Comment (nzardosh) [reply machine_learning_hep/processerdhadrons_jet.py:430]: but this cut should bee applied regardless of the pT right?
